src/phylum/init/sig.py

```diff
@@ -1,118 +1,83 @@
-functionality at the time of this writing.
+This module is meant to be a simple means of verifying RSA signatures in Python. It makes use of the hazardous materials
+layer of the `cryptography` library, but does so in a way that closely follows the example documentation:
 
-Short of forking the `py-minisign` repo to maintain it and publish a package from it on PyPI,
-the actual format for signatures and public keys is simple and so is verifying signatures.
+https://cryptography.io/en/latest/hazmat/primitives/asymmetric/rsa/#verification
 
-Minisign reference: https://jedisct1.github.io/minisign/
+There are a number of assumptions:
 
-Even still, this module is NOT meant to be used as a library or general purpose minisign
-signature verification. It is purpose written to specifically verify minisign signatures that
-were created by Phylum. As such, it makes a number of assumptions:
-
-* The Minisign Public Key for Phylum, Inc. will not change between releases
+* The RSA Public Key for Phylum, Inc. will not change between releases
+* The RSA signature was created with the SHA256 hash function
+* The RSA signature was created with the PKCS1 v1.5 padding scheme
 * The files to be verified were created by Phylum, Inc.
-* The `.minisig` signature includes a trusted comment and will therefore contain a known number of lines
-* The source of the `.minisig` signature is a trusted location, controlled by Phylum, Inc. for it's CLI releases
-"""
-import base64
-
-from cryptography.exceptions import InvalidSignature, UnsupportedAlgorithm
-from cryptography.hazmat.primitives.asymmetric import ed25519
-
-# This is the Minisign Public Key for Phylum, Inc. The matching private key was used to sign the software releases
-PHYLUM_MINISIGN_PUBKEY = "RWT6G44ykbS8GABiLXrJrYsap7FCY77m/Jyi0fgsr/Fsy3oLwU4l0IDf"
+* The source of the `.signature` files is a trusted location, controlled by Phylum, Inc. for it's CLI releases
 
-# The format for a minisign public key is:
-#
-# base64(<signature_algorithm> || <key_id> || <public_key>)
-#
-# signature_algorithm: `Ed`
-# key_id: 8 random bytes
-# public_key: Ed25519 public key
-PHYLUM_MINISIGN_PUBKEY_SIG_ALGO = base64.b64decode(PHYLUM_MINISIGN_PUBKEY)[:2]
-PHYLUM_MINISIGN_PUBKEY_KEY_ID = base64.b64decode(PHYLUM_MINISIGN_PUBKEY)[2:10]
-PHYLUM_MINISIGN_PUBKEY_ED25519 = base64.b64decode(PHYLUM_MINISIGN_PUBKEY)[10:]
+If these assumptions are not met, the signature verification will fail and the CLI install will exit with a message and
+a non-zero return code. The Phylum RSA public key is hard-coded in this module on purpose. It helps to limit network
+calls to GitHub, which can be a source of failure. It also has the advantage of "spreading" the public key to multiple
+locations so that a change to it (malicious or benign) will require access and coordination to each of those sources.
+It is understood that this method is not fool proof but should help to identify failures to the Phylum devs sooner.
```

Comment:
"help to identify failures to the Phylum devs" sounds a little strange, but I think it's still within reason.

I'd probably say it helps the Phylum devs identify failures or something like that if you want to change it.

Reply:
Changing it now...